src/Event/EventManagerTrait.php

```diff
@@ -17,66 +17,10 @@
 /**
  * Provides the event manager interface features for usage in classes that require it.
  *
+ * @deprecated 3.0.10 Use Cake\Event\EventDispatcherTrait instead.
  */
 trait EventManagerTrait
 {
 
-    /**
-     * Instance of the Cake\Event\EventManager this object is using
-     * to dispatch inner events.
-     *
-     * @var \Cake\Event\EventManager
-     */
-    protected $_eventManager = null;
-
-    /**
-     * Default class name for new event objects.
-     *
-     * @var string
-     */
-    protected $_eventClass = '\Cake\Event\Event';
-
-    /**
-     * Returns the Cake\Event\EventManager manager instance for this object.
-     *
-     * You can use this instance to register any new listeners or callbacks to the
-     * object events, or create your own events and trigger them at will.
-     *
-     * @param \Cake\Event\EventManager|null $eventManager the eventManager to set
-     * @return \Cake\Event\EventManager
-     */
-    public function eventManager(EventManager $eventManager = null)
-    {
-        if ($eventManager !== null) {
-            $this->_eventManager = $eventManager;
-        } elseif (empty($this->_eventManager)) {
-            $this->_eventManager = new EventManager();
-        }
-        return $this->_eventManager;
-    }
-
-    /**
-     * Wrapper for creating and dispatching events.
-     *
-     * Returns a dispatched event.
-     *
-     * @param string $name Name of the event.
-     * @param array|null $data Any value you wish to be transported with this event to
-     * it can be read by listeners.
-     * @param object|null $subject The object that this event applies to
-     * ($this by default).
-     *
-     * @return \Cake\Event\Event
-     */
-    public function dispatchEvent($name, $data = null, $subject = null)
-    {
-        if ($subject === null) {
-            $subject = $this;
-        }
-
-        $event = new $this->_eventClass($name, $subject, $data);
-        $this->eventManager()->dispatch($event);
-
-        return $event;
-    }
+    use EventDispatcherTrait;
```

Comment:
@jadb It could be, and actually was, but I think its less confusing that way. I imagine that somebody could be surprised if a file disappeared after 3.0.9 -> 3.0.10 bugfix update.